shared/cache.ts

```diff
@@ -0,0 +1,70 @@
+import { DateTime } from "luxon";
+
+type Cache<DataType> = {
+  value?: DataType;
+  updatedAt?: string;
+  timeToLive: number;
+  name?: string;
+};
+
+const defaultTimeToLive = 5 * 60;
+
+export const createCache = <DataType>(
+  getValue: () => Promise<DataType>,
+  defaultCache: Partial<Cache<DataType>> = {}
+) => {
+  let cache: Cache<DataType> = {
+    value: defaultCache.value ?? undefined,
+    updatedAt: defaultCache.updatedAt ?? undefined,
```

Comment:
Would we want to provide this from the consumer?

Reply:
I think if we're pre-populating the cache (ie default value) we'd want to set a default 'updatedAt', so we can set when the cache will expire 🙏 Maybe 'updatedAt' doesn't accurately describe it?

Otherwise, we could set the "default" to be more of a "config" with just `key` and `ttl`

Reply:
`populatedAt` which you have renamed to works, maybe it could be `populatedAtUtc` if we we want to be more specific?